File: 이건희/level3/정수 삼각형.py
def solution(triangle):
    dp = [[0] * i for i in range(1, len(triangle) + 1)]
    dp[0][0] = triangle[0][0]
    
    for i in range(len(triangle)):
        for j in range(i):
            dp[i][j] = max(dp[i][j],dp[i-1][j] + triangle[i][j])
            dp[i][j+1] = dp[i-1][j] + triangle[i][j+1]
            
    return max(dp[-1])

File: 이훈일/Programmers/Level3/정수 삼각형.java
import java.util.Arrays;

class Solution {
    public int solution(int[][] triangle) {
        for(int i = 1; i < triangle.length; i++) {
            for(int j = 0; j < triangle[i].length; j++) {
                if(j != 0 && j != i) {
                    triangle[i][j] += Math.max(triangle[i-1][j-1], triangle[i-1][j]); 
                }else if(j == 0) {
                    triangle[i][j] += triangle[i-1][j];
                }else if(j == i){
                    triangle[i][j] += triangle[i-1][j-1];
                }
            }
        }
        int[] lastArray = triangle[triangle.length-1];
        Arrays.sort(lastArray);
        return lastArray[lastArray.length-1];
    }
}
